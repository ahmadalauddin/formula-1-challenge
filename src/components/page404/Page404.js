import React from "react";
import { Link } from "react-router-dom";

import NotFound from "../../assets/images/not_found.svg";

const Page404 = () => {
  return (
    <>
    <div className="d-flex flex-column justify-content-center align-items-center">
      <img style={{maxWidth: 554}} src={NotFound} className="w-100-p" alt="Page Not Found" />
      <p>
        This is not the page that you are looking for!
      </p>
      <Link type="button" className="btn btn-outline-secondary" to="/">Go Home</Link>
      </div>
    </>
  );
};

export default Page404;
