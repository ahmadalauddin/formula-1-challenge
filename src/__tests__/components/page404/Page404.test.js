import React from "react";
import {shallow} from "enzyme";
import Page404 from "../../../components/page404/Page404";

let wrapper;

beforeEach(()=>{
  wrapper = shallow(<Page404/>);
  
})

describe("Testing champions list component", () => {
  it("Checking the snapshot of the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Checking if button is displayed properly", () => {
    expect(wrapper.find('Link').text()).toContain("Go Home");
  });

  it("Checking if not found image is displayed properly", () => {
    expect(wrapper.find('img').prop('src') ).toBe('not_found.svg');
  });

  it("Checking if the right text message is displayed", () => {
    expect(wrapper.find('p').text()).toBe('This is not the page that you are looking for!');
  });


});
