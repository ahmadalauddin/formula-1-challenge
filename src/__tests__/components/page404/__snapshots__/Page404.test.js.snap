// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing champions list component Checking the snapshot of the component 1`] = `
<Fragment>
  <div
    className="d-flex flex-column justify-content-center align-items-center"
  >
    <img
      alt="Page Not Found"
      className="w-100-p"
      src="not_found.svg"
      style={
        Object {
          "maxWidth": 554,
        }
      }
    />
    <p>
      This is not the page that you are looking for!
    </p>
    <Link
      className="btn btn-outline-secondary"
      to="/"
      type="button"
    >
      Go Home
    </Link>
  </div>
</Fragment>
`;
